onnxruntime/core/optimizer/nchwc_transformer.cc

```diff
@@ -783,6 +784,47 @@ void NchwcTransformerImpl::TransformBatchNormalization(Node& node) {
   removed_nodes_.push_front(node.Index());
 }
 
+void NchwcTransformerImpl::TransformTranspose(Node& node) {
+  auto& input_defs = node.MutableInputDefs();
+  auto& output_defs = node.MutableOutputDefs();
+
+  const ONNX_NAMESPACE::AttributeProto* perm_attr = graph_utils::GetNodeAttribute(node, "perm");
+  if (perm_attr == nullptr || perm_attr->ints_size() != 4) {
+    return;
+  }
+
+  // Test if this transposes from NCHW to NHWC layout order.
+  const int64_t* perm_data = perm_attr->ints().data();
+  if (perm_data[0] != 0 || perm_data[1] != 2 || perm_data[2] != 3 || perm_data[3] != 1) {
+    return;
+  }
+
+  // Don't transform the node if the input is not already in NCHWc format.
+  auto it = nchwc_args_.find(input_defs[0]);
+  if (it == nchwc_args_.end()) {
+    return;
+  }
+  auto* nchwc_input = it->second.get();
```

Comment:
nit: could do this before checking the perm attribute.

Reply:
That's fair. I did do it in the other order for TransformBatchNormalization and I do like consistency. I'll swap it in the next change to the file.